src/lib/sql/sql_pipeline.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <string>
+
+#include "SQLParserResult.h"
+#include "concurrency/transaction_context.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "sql/sql_query_plan.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+class SQLPipeline : public Noncopyable {
```

Comment:
Can we please get some tests for the pipeline? These should include failures both in the plan creation / execution as well as transaction aborts due to MVCC conflicts.

Reply:
@mrks Do you have a good test example for creation/execution failure? All I have at the moment is an invalid sql statement that fails during parsing and thus propagates the error all the way up to plan/execution. But that is not a plan/execution-specific error.

Reply:
load src/test/tables/int_float.tbl t
SELECT a+b FROM t